core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp

```diff
@@ -410,6 +410,11 @@ class ParallelFor<FunctorType, Kokkos::TeamPolicy<Properties...>,
     sycl::queue& q = *instance.m_queue;
 
     auto parallel_for_event = q.submit([&](sycl::handler& cgh) {
+      // Avoid capturing *this since it might not be trivially copyable
+      const auto shmem_begin     = m_shmem_begin;
+      const int scratch_size[2]  = {m_scratch_size[0], m_scratch_size[1]};
+      void* const scratch_ptr[2] = {m_scratch_ptr[0], m_scratch_ptr[1]};
```

Comment:
Now that it is a soon-to-be captured local var, why not make `scratch_ptr` a `char*` to avoid casting later?

Reply:
Note that this hasn't changed in this pull request. Anyway, I changed it now in 0d64ed38460f7e204b398f8b7bb0d4daf3620a20.